### ISSUE TITLE
Internal compiler error when using `emscripten_webgl_enable_extension` with WASM workers

### ISSUE THREAD

**Issue by edubart** [emscripten-core/emscripten#20134]
I get internal compiler error when I compiled the following example:

```c
#include <emscripten/html5.h>

int main() {
    emscripten_webgl_enable_extension(0, "WEBKIT_WEBGL_compressed_texture_pvrtc");
}
```

Output:
```
➜  emcc a.c -o a.html -sFULL_ES3 -sWASM_WORKERS=1
Internal compiler error in src/compiler.js!
Please create a bug report at https://github.com/emscripten-core/emscripten/issues/
with a log of the build and the input files used to run. Exception message: "Error: Invalid proxyingMode emscripten_webgl_enable_extension__proxy: 'sync_on_webgl_context_handle_thread' specified!
    at processLibraryFunction (/usr/lib/emscripten/src/jsifier.js:232:17)
    at addFromLibrary (/usr/lib/emscripten/src/jsifier.js:454:19)
    at symbolHandler (/usr/lib/emscripten/src/jsifier.js:554:16)
    at runJSify (/usr/lib/emscripten/src/jsifier.js:635:5)
    at Object.<anonymous> (/usr/lib/emscripten/src/compiler.js:102:3)
    at Module._compile (node:internal/modules/cjs/loader:1233:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1287:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
emcc: error: '/usr/bin/node /usr/lib/emscripten/src/compiler.js /tmp/tmp5966nagm.json' failed (returned 1)

➜  emcc -v
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.44-git (bec42dac7873903d09d713963e34020c22a8bd2d)
clang version 17.0.0 (/startdir/llvm-project a8cbd27d1f238e104a5d5ca345d93bc1f4d4ab1f)
Target: wasm32-unknown-emscripten
Thread model: posix
InstalledDir: /opt/emscripten-llvm/bin
```

My workaround is not use `emscripten_webgl_enable_extension`, I guess it is not supported yet with `sWASM_WORKERS` ?

**Comment by sbc100** [comment]
`WASM_WORKERS` today doesn't support any kind of automatic proxying of JS functions, so yes that function would not be supported.

However, you should see a better error message than this.  I idea is that should should see this error at runtime I think:

https://github.com/emscripten-core/emscripten/blob/86fa8e58b28cd9c3ee53d953d40e2f7ab5308446/src/jsifier.js#L252

We should probably fix it so that you don't get a built time error.

**Comment by sbc100** [comment]
Actually `emscripten_webgl_enable_extension` should work fine as long as you call it from the main thread, so this build error is certainly a bug.  Fix is in #20135
